### PR TITLE
SecretFileServer stores all key, cert, root and add unit tests.

### PR DESCRIPTION
**Pull request by incfly**
- Change the SecretServer interface to `Put(serviceAccount string, KeyCertBundle), Get(string)`. So caller will provide the serviceAccount, instead of doing certificate & string parsing to extract the service account value.
- Add unit test to verify file has expected name and content for each service account.

### CODE REVIEW

File: security/pkg/workload/secretfileserver.go [hunk 1]
```diff
@@ -34,15 +36,28 @@ type SecretFileServer struct {
...
+	if err := ioutil.WriteFile(cpath, cert, CertFilePermission); err != nil {
 		return err
 	}
-	return ioutil.WriteFile(kpath, cert, CertFilePermission)
+	rpath := path.Join(dir, "root-cert.pem")
+	return ioutil.WriteFile(rpath, root, CertFilePermission)
+}
+
+// Get is not implemented on purpose since workload pods get key cert by hostPath mounts.
+func (sf *SecretFileServer) Get(serviceAccount string) (util.KeyCertBundle, error) {
```
Comment (wattli) [inline security/pkg/workload/secretfileserver.go]: I assume we need 'Get()' here is to follow the SecretServer interface. What about move Get() out of the interface? And only implement it at the instance that needs it.

Comment (incfly) [reply security/pkg/workload/secretfileserver.go]: Done.

Once we implemented SDS server, no longer need file based implementation, could just delete interface by then.

Comment (wattli) [reply security/pkg/workload/secretfileserver.go]: Then we don't need this function any more, right?

File: security/pkg/workload/server.go [hunk 2]
```diff
@@ -90,11 +90,16 @@ func (s *SDSServer) SetServiceIdentityPrivateKey(content []byte) error {
...
 
-// Save saves the specified key cert.
-func (s *SDSServer) Save(b util.KeyCertBundle) error {
+// Put stores the KeyCertBundle for a specific service account.
+func (s *SDSServer) Put(serviceAccount string, b util.KeyCertBundle) error {
 	return nil
 }
 
+// Get returns the KeyCertBundle for a specific service account.
+func (s *SDSServer) Get(serviceAccount string) (util.KeyCertBundle, error) {
```
Comment (wattli) [inline security/pkg/workload/server.go]: and we don't need this function for now as well, right?

Comment (incfly) [reply security/pkg/workload/server.go]: yes. not needed for now.

File: security/pkg/workload/secretfileserver.go (hunk 1 above)
Comment (incfly) [reply security/pkg/workload/secretfileserver.go]: Deleted Get from two structs.

**Pull request merged**